src/SoundBank.js

```diff
@@ -0,0 +1,84 @@
+const SoundPlayer = require('./GreenPlayer');
+const EffectsChain = require('./EffectsChain');
+
+const ALL_TARGETS = '*';
+
+class SoundBank {
+    constructor (audioEngine) {
+        this.audioEngine = audioEngine;
+
+        this.soundPlayers = {};
+        this.playerTargets = new Map();
+        this.soundEffects = new Map();
+    }
+
+    getSound (soundId) {
+        if (!this.soundPlayers[soundId]) {
+            this.soundPlayers[soundId] = new SoundPlayer(this.audioEngine, {
+                id: soundId, buffer: this.audioEngine.audioBuffers[soundId]
+            });
+        }
+
+        return this.soundPlayers[soundId];
+    }
+
+    getSoundEffects (sound) {
+        if (!this.soundEffects.has(sound)) {
+            this.soundEffects.set(sound, new EffectsChain(this.audioEngine));
+        }
+
+        return this.soundEffects.get(sound);
+    }
+
+
+    playSound (target, soundId) {
+        const effects = this.getSoundEffects(soundId);
+        const player = this.getSoundPlayer(soundId);
+
+        this.playerTargets.set(soundId, target);
+        effects.setFromTarget(target);
+        effects.addSoundPlayer(player);
+
+        player.connect(effects);
+        player.play();
+
+        return player.finished();
+    }
+
+    setEffects (target) {
+        this.playerTargets.forEach((playerTarget, key) => {
+            if (playerTarget === target) {
+                this.getSoundEffects(key).setFromTarget(target);
+            }
+        });
+    }
+
+    stop (target, soundId) {
+        if (this.playerTargets.get(soundId) === target) {
+            this.soundPlayers[soundId].stop();
```

Comment:
should probaby use `this.getSound(soundId)` here for future refactor possibilities.